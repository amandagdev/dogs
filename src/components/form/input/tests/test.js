import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'

import Input from '..'
import userEvent from '@testing-library/user-event'

describe('<Input>', () => {
  it('should render the input', () => {
    const { container } = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('must simulate an onChange', async () => {
    const onInput = jest.fn()

    render(<Input label="name" id="name" labelFor="name" onChange={onInput} />)

    const input = screen.getByRole('textbox')
    const text = 'this is a teste'
    userEvent.type(input, text)
    expect(onInput).toHaveBeenCalledTimes(text.length)
  })
})
