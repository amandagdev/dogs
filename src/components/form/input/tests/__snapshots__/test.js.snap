// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Input> should render the input 1'] = `
<div
  class="input__wrapper"
>
  <label
    class="input__label"
  />
  <input
    class="input"
    value=""
  />
  <p
    class="input__error"
  >
    Error
  </p>
</div>
`
